inc/render-partials.php

```diff
@@ -0,0 +1,192 @@
+	}
+}
+
+/**
+ * Render the 'Contribute' button
+ *
+ * Takes the contribute_url setting and coditionally renders the button.
+ */
+function uds_wp_render_contribute_button() {
+	$contribute_url = get_theme_mod( 'contribute_url' );
+	$contribute_template = '<p class="contribute-button"><a href="%s" type="button" class="btn btn-gold">Contribute</a></p>';
+
+	// Do we have a contribute?
+	if ( $contribute_url && '' !== $contribute_url ) {
+		echo wp_kses( sprintf( $contribute_template, $contribute_url ), wp_kses_allowed_html( 'post' ) );
+	}
+}
+
+/**
+ * Render Footer branding row (logo and social media icons together)
+ *
+ * Determines whether or not we render the logo+social row of the footer.
+ * When this is set to 'disabled' we skip generating the markup altogether,
+ * meaning that individual settings for logo or social icons are ignored.
+ */
+function uds_wp_render_footer_branding_row() {
+	$row_status = get_theme_mod( 'footer_row_branding' );
+
+	if ( 'enabled' === $row_status ) {
+		?>
+		<div class="container" id="endorsed-footer">
+			<div class="row">
+
+				<div class="col-md" id="endorsed-logo">
+					<?php uds_wp_render_footer_logo(); ?>
+				</div>
+
+				<div class="col-md" id="social-media">
+					<div class="social-media-wrapper">
+						<?php
+						if ( has_nav_menu( 'social-media' ) ) {
+							wp_nav_menu(
+								array(
+									'theme_location'  => 'social-media',
+									// 'container' => 'div',
+									// 'container_class' => 'col-md',
+									// 'container_id' => 'social-media',
+									'menu_class'  => '',
+									'items_wrap' => '<nav aria-label="Social Media" class="nav">%3$s</nav>',
+									'walker' => new WP_Social_Media_Walker(),
+								)
+							);
+						}
+						?>
+					</div>
+				</div>
+			</div> <!-- row -->
+		</div> <!-- endorsed-footer -->
+		<?php
+	}
+}
+
+/**
+ * Render 'Contact Us' link in the footer
+ *
+ * Takes the value of the 'contact_url' setting and, if it is not empty,
+ * renders a link with that value as the URL.
+ */
+function uds_wp_render_contact_link() {
+	// =============================
+	$contact_url = get_theme_mod( 'contact_url' );
+	$contact_template = '<p class="contact-link"><a href="%1$s">Contact Us</a></p>';
+
+	if ( $contact_url && '' !== $contact_url ) {
+		echo wp_kses( sprintf( $contact_template, $contact_url ), wp_kses_allowed_html( 'post' ) );
+	}
+}
+
+
+/**
+ * Render the info+menu links row in the footer
+ *
+ * Determines whether or not we render the info+menu row of the footer.
+ * When this is set to 'disabled' we skip generating the markup altogether,
+ * meaning that individual settings for contact, contribute, and the menu
+ * items are ignored altogether.
+ */
+function uds_wp_render_footer_action_row() {
+	$action_row_status = get_theme_mod( 'footer_row_actions' );
+	$org_name = wp_kses( get_bloginfo( 'name' ), array() );
```

Comment:
Wrapping get_blogingo() in wp_kses() is unnecessary. get_bloginfo() supports filtering the property being retrieved (though 'raw' is the default filter option.)

See [get_bloginfo Function Page](https://developer.wordpress.org/reference/functions/get_bloginfo/)

Reply:
But it is a good question for us to ask. Do we want/need to filter our Site Name property to prevent embedded html tags? If so, then we should set the filter option here and wherever we use get_bloginfo('name'). We have this inc/render-partials.php (line 23).

And we should probably reuse that render function here anyway.